src/include/migraphx/op/scatternd_op.hpp

```diff
@@ -28,44 +28,87 @@
 #include <migraphx/check_shapes.hpp>
 #include <migraphx/argument.hpp>
 #include <migraphx/par_for.hpp>
+#include <migraphx/ranges.hpp>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace op {
-
+/**
+ * @brief
+ * N-dimensional Scatter operations. This struct is parent class to ops which differ in what formula
+ * is used to reduce (combine old and new values of) the scattered value.  It is an implementation
+ * of the Onnx ScatterND operation (see
+ * https://github.com/onnx/onnx/blob/main/docs/Operators.md#ScatterND)
+ *
+ * @tparam Derived a type erasure parameter represents one of the child operations.
```

Comment:
There is no type erasure here. This is just using CRTP: https://en.cppreference.com/w/cpp/language/crtp

Reply:
comment changed.